### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by ezraermy**
In this pull request, we: 
- Created data_manager.rb which provides methods for saving and loading data related to books, people, and rentals.
- save_data method is the main entry point that triggers the saving of books, people, and rentals.
- load_data method is the main entry point for loading data and calls the load_books, load_people, and load_rentals methods.
### Other requirements
- No linter errors
- Follow GitHub flow
- Document professional way.

### CODE REVIEW

File: data_manager.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                  else
+                    teacher = Teacher.new(person_info['age'], person_info['specialization'], person_info['name'])
+                    teacher.role = 'teacher'
+                    teacher
+                  end
+                end
+              end
+  end
+
+  def load_rentals
```
Comment (amejid) [inline data_manager.rb:75]: - [ ] You did a fantastic job persisting books and people. :+1: There are issues when rental data is loaded from the file. After creating a rental for a particular user and verifying it exists, once the app is restarted rental data seems to get lost. There is a problem displaying rentals belonging to a particular person after restarting the app. Please work on this issue. 🤝️

**Review by amejid — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Data getting persisted using JSON files :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @ezraermy 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ezraermy** [comment]
Hi @amejid and @omarsalem7. Thank you both for your helpful feedback.

**Pull request merged**